pkg/flavor/flavor.go

```diff
@@ -43,95 +44,116 @@ func NewMethods(flavorStore, clusterFlavorStore cache.Store) Methods {
 	}
 }
 
-func (m *methods) FindProfile(vm *virtv1.VirtualMachine) (*flavorv1alpha1.VirtualMachineFlavorProfile, error) {
+func (m *methods) FindAndApplyFlavor(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) (Conflicts, error) {
+
 	if vm.Spec.Flavor == nil {
 		return nil, nil
 	}
 
-	profiles, err := getProfiles(vm.Spec.Flavor.Name, vm.Namespace, vm.Spec.Flavor.Kind, m.flavorStore, m.clusterFlavorStore)
+	flavorSpec, err := m.FindFlavorSpec(vm)
 	if err != nil {
 		return nil, err
 	}
 
-	if vm.Spec.Flavor.Profile == "" {
-		profile := findFirstProfile(profiles, func(profile *flavorv1alpha1.VirtualMachineFlavorProfile) bool {
-			return profile.Default
-		})
-		if profile == nil {
-			return nil, fmt.Errorf("flavor does not specify a default profile")
-		}
-		return profile, nil
-	} else {
-		profile := findFirstProfile(profiles, func(profile *flavorv1alpha1.VirtualMachineFlavorProfile) bool {
-			return profile.Name == vm.Spec.Flavor.Profile
-		})
-		if profile == nil {
-			return nil, fmt.Errorf("flavor does not have a profile with name: %v", vm.Spec.Flavor.Profile)
-		}
-		return profile, nil
+	if flavorSpec == nil {
+		return nil, nil
 	}
-}
 
-func (m *methods) ApplyToVmi(field *k8sfield.Path, profile *flavorv1alpha1.VirtualMachineFlavorProfile, vmiSpec *virtv1.VirtualMachineInstanceSpec) Conflicts {
-	var conflicts Conflicts
+	conflicts := apply(
+		k8sfield.NewPath("spec", "template", "spec"),
+		flavorSpec,
+		vm,
+		vmi,
+	)
 
-	conflicts = append(conflicts, applyCpu(field, profile, vmiSpec)...)
+	return conflicts, nil
 
-	return conflicts
 }
 
-func findFirstProfile(profiles []flavorv1alpha1.VirtualMachineFlavorProfile, predicate func(profile *flavorv1alpha1.VirtualMachineFlavorProfile) bool) *flavorv1alpha1.VirtualMachineFlavorProfile {
-	for i := range profiles {
-		profile := &profiles[i]
-		if predicate(profile) {
-			return profile
+func (m *methods) FindFlavorSpec(vm *virtv1.VirtualMachine) (*flavorv1alpha1.VirtualMachineFlavorSpec, error) {
+
+	switch strings.ToLower(vm.Spec.Flavor.Kind) {
```

Comment:
Can you add a `case` for cluster flavors and `default` branch that returns an error, if the `Kind` does not match any case?

Reply:
Done.